### PR TITLE
Custom migration scripts

### LINKED ISSUE #56: Customize migrations for different setups
We have different setups for the whole system (ganache, parity or truffle develop). It is currently not supported to work out of the box on those setups while developing. To make it easier we should customize the `./migrations/2_deploy_contracts.js` file to our needs. It should handle the different setups and for example add accounts as households to the utility contract, etc. See https://truffleframework.com/docs/truffle/getting-started/running-migrations.

### PR DESCRIPTION
**Pull request by dakingha69**
Closes #56 

This includes migration scripts for `ganache` and `authority` setups. See README for how to run the whole system in the different setups.

One interesting thing to note here is that for some reason contracts that are specified in the chain spec of parity are deployed with a `msg.sender` of a `ZERO_ADDRESS`. So I had to customize and hard code the owner in the `Ownable` contract.

### CODE REVIEW

**Comment by myronrotter** [comment]
@dakingha69 after the UC refactoring yesterday, we need to adjust this setup once more. Im sorry for the inconvenience. I had a quick look into this PR and this is what I think the refactoring broke:

- `Blockreward.sol` calls `settle` of `FifsUtility.sol`, now `Utility.sol`
- `migrations/2_deploy_contracts.js` needs to migrate `UtilityBase.sol` and `Utility.sol`
- Small things in README

There might be more.

**Comment by dakingha69** [comment]
@myronrotter no worries all good! incorporated the refactored contracts

File: household-server/transaction-handler.js [hunk 1]
```diff
@@ -32,21 +19,19 @@ module.exports = {
...
-      await web3.eth.personal.unlockAccount(address, password, 600);
+      await web3.eth.personal.unlockAccount(address, password, null);
       const txReceipt = await contract.methods
         .updateRenewableEnergy(
           address,
           conversionHelper.kWhToWs(produce),
           conversionHelper.kWhToWs(consume)
         )
-        .send({ from: address });
+        .send({ from: address, gasLimit: 16179379 });
```
[the remark this answers is not in the document]
Comment (dakingha69) [reply household-server/transaction-handler.js]: Ups yea... was a quick fix tbh. Parity was complaining about too high gas limit when sending the tx.

**Comment by dakingha69** [comment]
@paguos thx for review! the merge conflict should be fixed now

**Comment by paguos** [comment]
Now I get the following error:

```sh
2_deploy_contracts.js
=====================
Transaction: 0x0d84c8ee66309be3cc1787d03155be6100203466effa8abcc34efe8f5fbe35f8 exited with an error (status 0). 
     Please check that the transaction:
     - satisfies all conditions set by Solidity `require` statements.
     - does not trigger a Solidity `revert` statement.

Truffle v5.0.14 (core: 5.0.14)
Node v10.15.3

```

**Comment by dakingha69** [comment]
hmm did you restart the docker containers?

**Comment by paguos** [comment]
yes sir did a cleanup before running it again 😉

**Comment by dakingha69** [comment]
weird.... also after running `sudo docker-compose down -v`?

**Comment by paguos** [comment]
Yep

**Comment by paguos** [comment]
@dakingha69 I try it again but failed. Having the same problem deploying the validator set, when invoking the `addValidator` method... 

In your PR it doesn't even deploy the migration contract :/

**Comment by paguos** [comment]
@dakingha69 I believe this is mac related (or at least my computer) problem ... Like I'm mentioned before I'm having the same issues. I went again through your code and couldn't find anything wrong. 

- It finds the utility contract but when it tries to invoke a method it fails
- I had the same problem using truffle or just web3js

For my side, this PR is more than fine 👍 🥇  Will bother you to test the `ValidatorSet` solutions. Because it seems I've been wasting time trying to make it run on my machine 🤕

**Pull request merged**